core/src/server/rpc_module.rs

```diff
@@ -1110,29 +1111,29 @@ impl SubscriptionSink {
 	/// }
 	/// ```
 	///
-	pub fn close(self, err: impl Into<ErrorObjectOwned>) -> bool {
+	pub fn close(self, err: impl Into<ErrorObjectOwned>) -> impl Future<Output = ()> {
```

Comment:
Since the subscription will gracefully close if you don't call this message too, and this is really just a way to close the subscription with some error, I wonder whether it should be called eg `close_with_error`?

Reply:
yeah, it's a way to indicate to the connected client that subscription is closed with a special notification.

if one doesn't call this at least our client will wait until the connection is closed unless the subscription itself has customized logic when a subscription is regarded as "closed"

Reply:
thus, this may be an error or success.

but maybe we should have `close_with_success` and `close_with_error` instead as you suggested.